imap/update_message_labels_updated.go

```diff
@@ -11,29 +11,34 @@ type MessageMailboxesUpdated struct {
 
 	*updateWaiter
 
-	MessageID            MessageID
-	MailboxIDs           []MailboxID
-	Seen, Flagged, Draft bool
+	MessageID  MessageID
+	MailboxIDs []MailboxID
+
+	CustomFlags MessageCustomFlags
+}
+
+type MessageCustomFlags struct {
+	Seen, Flagged, Draft, Answered bool
+}
+
+func (f MessageCustomFlags) String() string {
+	return fmt.Sprintf("seen=%v, flagged=%v, draft=%v, answered=%v", f.Seen, f.Flagged, f.Draft, f.Answered)
 }
 
-func NewMessageMailboxesUpdated(messageID MessageID, mailboxIDs []MailboxID, seen, flagged, draft bool) *MessageMailboxesUpdated {
+func NewMessageMailboxesUpdated(messageID MessageID, mailboxIDs []MailboxID, flags MessageCustomFlags) *MessageMailboxesUpdated {
 	return &MessageMailboxesUpdated{
 		updateWaiter: newUpdateWaiter(),
 		MessageID:    messageID,
 		MailboxIDs:   mailboxIDs,
-		Seen:         seen,
```

Comment:
While these changes are correct. I think it would be better if we normalized the flag manipulation interface so that we can send any type of flag back and forth.